gnosis/cowswap
# [1283 pt.1] Add Matcha-0x Price source

# Summary

Part of #1283 

This PR adds Matcha-0x as a price source following the configuration created by @josojo in https://github.com/gnosis/gp-v2-services/blob/main/solver/src/solver/matcha_solver/api.rs#L5

This PR moves a few files and required path changes = lots of noise. The main changed files are:
1. Match-0x price api: https://github.com/gnosis/cowswap/pull/1312/files#diff-8e927d416fe79ec99997dc4c25ef12addc58f1b6a40161857f8a903bf0c7f2ae
2. Prices base file using all the prices sources: https://github.com/gnosis/cowswap/pull/1312/files#diff-ace987d672e598e88fce90795b25a0e54b34950d91cbe9757b38f3e945d2932f

Justification behind moving files: the `utils` folder was quickly getting oversaturated with files, many not even "utility" functions and more core api files. What was previously `utils > operator > etc`is now:
```
api--
   gnosisProtocol--
         index.ts
         gnosisProtocolApi.ts
         ...etc
   paraswap--
         index.ts
   match-0x--
         index.ts
```

## Screenshots
Logs showing "winning price"
![image](https://user-images.githubusercontent.com/21335563/130781244-376c9702-9c1f-422d-85a5-6c4de6cd005d.png)

Logs showing prices
![image](https://user-images.githubusercontent.com/21335563/130781408-9d7dc67c-183d-4b1b-b3db-b2a6d7e039a0.png)

  # To Test
_api only available in Mainnet and Ropsten_
1. test swapping between tokens
2. check logs
3. check for the `Error: try again later` issue is as common
